### ISSUE TITLE
Enable examples tests

### ISSUE THREAD

**Issue by stoyanK7** [checkstyle/checkstyle#13345]
This Issue deals with enabling Example Checks.
If you plan to help us to fix issue, thanks a lot first of all, and secondly please make a comment "I am on xxxCheck" to let others know that you are working on it to avoid conflicts.

Some examples from the website are uncompilable and untested ..... conversion of `.txt` to `.java` and removal of `@Disabled `annotatioon in Test class will make them active.

Examples of ideal updates : 
- https://github.com/checkstyle/checkstyle/commit/7123cc2b7ca95f5f349e54776a02724f2e7b0d69
- https://github.com/checkstyle/checkstyle/commit/ddbcf447682a58a9de237391dad99e96903cb969
- https://github.com/checkstyle/checkstyle/commit/2cec4d51ffb0478689039251c66a4e0253d6ec9f

These commits implement your idea of `example same by java code and different by violation comments` .

Please create 1 PR for each Check , please look at referenced PR commits as example of update.

**_RULES :_**

- The java code/structure is same for all examples ( _meaning they should follow one template and look the same_ )
- Only comments are changing in content from Example1 to Example2 and so on....
- If code line is present in one Example on Line X, it should be located on same line in all other Examples. 
- Lines raising violations should be commented using " `// violation` "
- If line contains only violation comment but violation is not raised, it should be empty line or " `// ok, because ....` "

Visual sample of how examples should be same by java structure/statements but different by comments 
([taken from PR](https://github.com/checkstyle/checkstyle/pull/14166/files)):
![image](https://github.com/checkstyle/checkstyle/assets/812984/db385c6a-2957-460a-b83a-bd3f7fd90f8c)
 one more example([html](https://checkstyle.org/checks/naming/staticvariablename.html#Example1-code), [pull request](https://github.com/checkstyle/checkstyle/pull/14345/files)):
![image](https://github.com/checkstyle/checkstyle/assets/812984/e293b8d5-bcf0-419e-85b1-0aca357d35f7)
one more example (java code reuse same line numbers and only comments are changing by content):
![image](https://github.com/checkstyle/checkstyle/assets/812984/8c85ede3-fcc6-4b12-b692-2bbb1ee837dc)
  

List of remaining checks' tests:

 - [x] SuppressWithNearbyTextFilterExamplesTest.java
 - [x] SuppressWarningsFilterExamplesTest.java
 - [x] SuppressionSingleFilterExamplesTest.java
 - [x] SuppressionXpathSingleFilterExamplesTest.java
 - [x] SuppressWithNearbyCommentFilterExamplesTest.java
 - [x] SuppressWithPlainTextCommentFilterExamplesTest.java
 - [x] SuppressionCommentFilterExamplesTest.java
 - [x] RegexpSinglelineCheckExamplesTest.java
 - [x] RegexpSinglelineJavaCheckExamplesTest.java
 - [x] RegexpCheckExamplesTest.java
 - [x] RegexpOnFilenameCheckExamplesTest.java
 - [x] RegexpMultilineCheckExamplesTest.java
 - [x] PackageNameCheckExamplesTest.java
 - [x] ClassFanOutComplexityCheckExamplesTest.java
 - [x] CyclomaticComplexityCheckExamplesTest.java
 - [x] JavaNCSSCheckExamplesTest.java
 - [x] NPathComplexityCheckExamplesTest.java
 - [x] AnonInnerLengthCheckExamplesTest.java
 - [x] LineLengthCheckExamplesTest.java
 - [x] ExecutableStatementCountCheckExamplesTest.java
 - [x] InterfaceMemberImpliedModifierCheckExamplesTest.java
 - [x] TranslationCheckExamplesTest.java
 - [x] IndentationCheckExamplesTest.java
 - [x] CommentsIndentationCheckExamplesTest.java
 - [x] OrderedPropertiesCheckExamplesTest.java
 - [x] DescendantTokenCheckExamplesTest.java
 - [x] PackageDeclarationCheckExamplesTest.java
 - [x] MissingSwitchDefaultCheckExamplesTest.java
 - [x] NestedTryDepthCheckExamplesTest.java
 - [x] MagicNumberCheckExamplesTest.java
 - [x] HeaderCheckExamplesTest.java
 - [x] RegexpHeaderCheckExamplesTest.java
 - [x] UniquePropertiesCheckExamplesTest.java
 - [x] ImportOrderCheckExamplesTest.java
 - [x] IllegalImportCheckExamplesTest.java
 - [x] CustomImportOrderCheckExamplesTest.java
 - [x] JavadocMethodCheckExamplesTest.java
 - [x] SummaryJavadocCheckExamplesTest.java
 - [x] JavadocTypeCheckExamplesTest.java
 - [x] WriteTagCheckExamplesTest.java
 - [x] JavadocStyleCheckExamplesTest.java
 - [x] MissingJavadocMethodCheckExamplesTest.java
 - [x] JavadocPackageCheckExamplesTest.java
 - [x] MissingJavadocTypeCheckExamplesTest.java
 - [x] JavadocTagContinuationIndentationCheckExamplesTest.java
 - [x] MissingJavadocPackageCheckExamplesTest.java
 - [x] NonEmptyAtclauseDescriptionCheckExamplesTest.java
 - [x] BeforeExecutionExclusionFileFilterExamplesTest.java

------------------
Generate list of tests by:
```
DISABLED_EXAMPLE_TESTS=$(ag -Ql '@Disabled("until https://github.com/checkstyle/checkstyle/issues/13345")')

echo "$DISABLED_EXAMPLE_TESTS" | while read -r line; do b=$(basename "$line") && echo " - [ ] $b"; done

```

**Comment by nrmancuso** [comment]
From https://github.com/checkstyle/checkstyle/pull/13482#discussion_r1278314430:

> Just something to keep in mind: some of these will have to get changed to `getNonCompilablePath` if the examples use newer syntax. We should probably target one of these for an example in the issue.

**Comment by nrmancuso** [comment]
I am on SuperCloneCheckExamplesTest

**Comment by Guneetsinghtuli** [comment]
I am on MissingOverrideCheck

**Comment by Rohanraj123** [comment]
I am on  AnnotationUseStyleCheckExamplesTest.java

**Comment by suniti0804** [comment]
I am working on AnnotationUseStyleCheckExamplesTest.java

**Comment by suniti0804** [comment]
I am on EmptyStatementCheck

**Comment by Guneetsinghtuli** [comment]
I am on LeftCurlyCheck

**Comment by TanayMorakhia** [comment]
@stoyanK7 Can you please check off the ones that are merged. So that I can know which one I can work on.

**Comment by romani** [comment]
@TanayMorakhia, yes, we need to fix checkboxes, but to unblock you, please choose any check if it has Disabled annotation and usage of txt files in paths, if this found it is not migrated

**Comment by suniti0804** [comment]
I am on EmptyStatementCheck

**Comment by suniti0804** [comment]
I am on EmptyStatementCheck

**Comment by AayushSaini101** [comment]
I am on UpperEllCheckExamplesTest

**Comment by AayushSaini101** [comment]
I am on ParameterAssignmentCheckExamplesTest

**Comment by rnveach** [comment]
List of remaining items is at https://github.com/checkstyle/checkstyle/blob/d316b893da935d33736e90321b909f9b72d835d8/config/checkstyle-non-main-files-suppressions.xml#L170-L176

**Comment by RAHULDAS6009** [comment]
I am on  SuppressWarningsHolderExamplesTest

**Comment by TanayMorakhia** [comment]
I am on JavaNCSSCheckExamplesTest

**Comment by TanayMorakhia** [comment]
I am on AvoidStarImportCheckExamplesTest

**Comment by relentless-pursuit** [comment]
@RAHULDAS6009, would you mind if i send the PR. I have the PR almost ready.

**Comment by romani** [comment]
@relentless-pursuit , if no resoponse in 2 days, please send PR.

**Comment by relentless-pursuit** [comment]
> @relentless-pursuit , if no resoponse in 2 days, please send PR.

I have already sent a PR. I hope the contributor won't mind as there seeemed no action it.

**Comment by RAHULDAS6009** [comment]
> > @relentless-pursuit , if no resoponse in 2 days, please send PR.
> 
> I have already sent a PR. I hope the contributor won't mind as there seeemed no action it.

no problem i will take another issue

**Comment by RAHULDAS6009** [comment]
I am on  PackageAnnotationCheckExamplesTest

**Comment by TanayMorakhia** [comment]
I am on RedundantImportExampleTest

**Comment by suniti0804** [comment]
I am on NeedBracesCheck

**Comment by suniti0804** [comment]
I am on NeedBracesCheck

**Comment by piyush-1234** [comment]
I am on PackageAnnotationCheck

**Comment by piyush-1234** [comment]
I am on AvoidNestedBlocksCheck

**Comment by suniti0804** [comment]
I am on Local Variable Name Check

**Comment by suniti0804** [comment]
I am on EmptyBlockCheckExamplesTest

**Comment by piyush-1234** [comment]
i am on RightCurlyCheckExamplesTest.java

**Comment by suniti0804** [comment]
I am on ArrayTrailingCommaCheckExamplesTest

**Comment by piyush-1234** [comment]
i am on TypeNameCheckExamplesTest.java

**Comment by piyush-1234** [comment]
i am on LocalFinalVariableNameCheckExamplesTest.java

**Comment by piyush-1234** [comment]
i am on PackageAnnotationCheckExamplesTest.java

**Comment by piyush-1234** [comment]
i am on AvoidNestedBlocksCheckExamplesTest.java

**Comment by rishivardhanmm** [comment]
I am on MethodTypeParameterNameCheckExamplesTest.java

**Comment by Kai0011** [comment]
For some test examples, such as examples for EmptyCatchBlockCheck, the txt can not be converted into java because there is not class name, method name, only a try catch bloack. Could you please tell me how to deal with such cases? @romani @rnveach @nrmancuso @stoyanK7

**Comment by romani** [comment]
@Kai0011 , do not hesitate to add required code lines to make it compile able.
Look at example at  https://github.com/checkstyle/checkstyle/commit/2b86617a7e1e025cfcef7201ce5cad372da90d43#diff-405a078560131d52bcc2abb8df46be01eb0135e3b49d180da921b85df9da0bbdR42

Just add all required code to such files.

**Comment by TarunAlapati-git** [comment]
I am on  LambdaParameterNameCheckExamplesTest

**Comment by andreiyusupau** [comment]
Converting the Example files to .java for some checks like `HideUtilityClassConstructor` leads to failing test, because actual violations count is higher than expected (2 for actual and 0 for expected). How this could be fixed? Seems the problem is with `verifyWithInlineConfigParser` method:
```
Violation lines for C:\Users\andreiyusupau\IdeaProjects\personal\checkstyle\src\xdocs-examples\resources\com\puppycrawl\tools\checkstyle\checks\design\hideutilityclassconstructor\Example1.java differ.
expected to be empty
but was: [10, 35]

at com.puppycrawl.tools.checkstyle.AbstractModuleTestSupport.verifyViolations(AbstractModuleTestSupport.java:470)
	at com.puppycrawl.tools.checkstyle.AbstractModuleTestSupport.verifyWithInlineConfigParser(AbstractModuleTestSupport.java:238)
	at com.puppycrawl.tools.checkstyle.checks.design.HideUtilityClassConstructorCheckExamplesTest.testExample1(HideUtilityClassConstructorCheckExamplesTest.java:39)

```

**Comment by romani** [comment]
>How this could be fixed?

We just need to show in Example actual behavior, mark all places/lines as violation  that needs it.

**Comment by SinghaAnirban005** [comment]
I am on NeedBracesCheckExamplesTest

**Comment by piyush-1234** [comment]
i am on  RightCurlyCheckExamplesTest.java

**Comment by MauryanKansara** [comment]
I'm working on NewlineAtEndOfFileCheckExamplesTest.

**Comment by aayushRedHat** [comment]
I am on  NeedBracesCheckExamplesTest

**Comment by kapilgahlot1998** [comment]
I am working on CatchParameterNameCheckExamplesTest

**Comment by goyalg325** [comment]
I am on  HideUtilityClassConstructorCheck

**Comment by Coding-Aliens** [comment]
Working on  IllegalTokenTextCheckExamplesTest.java

**Comment by kapilgahlot1998** [comment]
Working on StaticVariableNameCheckExamplesTest

**Comment by Coding-Aliens** [comment]
Working on FileLengthCheckExamplesTest

**Comment by sktpy** [comment]
I am on InterfaceIsTypeCheckExamplesTest

**Comment by mahfouz72** [comment]
I am on  StaticVariableNameCheckExamplesTest

**Comment by mahfouz72** [comment]
I didn't notice that someone on this file @kapilgahlot1998.I see that your PR is a draft so are you working on this file or should I continue working on it

**Comment by mahfouz72** [comment]
Ok, I will take another check. I am on PatternVariableNameCheckExamplesTest

**Comment by mahfouz72** [comment]
I am on FinalParametersCheckExamplesTest.java

**Comment by mahfouz72** [comment]
I am on UnnecessaryParenthesesCheckExamplesTest

**Comment by mahfouz72** [comment]
I am on UnnecessaryParenthesesCheckExamplesTest

**Comment by mahfouz72** [comment]
I am on FinalParametersCheckExamplesTest

**Comment by MANISH-K-07** [comment]
I am on sizes module remaining checks..

**Comment by mahfouz72** [comment]
I am on  MethodLengthCheckExamplesTest

**Comment by ratatall** [comment]
Hey! Just wanted to tell that I'll be working on ThrowsCountCheckExamplesTest

**Comment by Lmh-java** [comment]
I am on EqualsHashCodeCheckExamplesTest

**Comment by lmgyuan** [comment]
I am on MutableExceptionCheckExamplesTest. However, I am a bit confused about how to do this. I am not able to find the .txt associated with MutableExceptionCheck to turn them into java. Is there anything that I misunderstood? Thanks in advance for any information!

**Comment by MANISH-K-07** [comment]
> I am on MutableExceptionCheckExamplesTest. However, I am a bit confused about how to do this. I am not able to find the .txt associated with MutableExceptionCheck to turn them into java. Is there anything that I misunderstood? Thanks in advance for any information!

I see Example files in txt format at https://github.com/checkstyle/checkstyle/tree/master/src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/design/mutableexception

**Comment by Lmh-java** [comment]
I am on `UnusedLocalVariableCheckExamplesTest`

**Comment by lmgyuan** [comment]
I am on OrderedPropertiesCheckExamplesTest

**Comment by Kexin2000** [comment]
I am on ParameterNumberCheckExamplesTest

**Comment by OmarShawky1** [comment]
I'm on AvoidNestedBlocksCheckExamplesTest

**Comment by romani** [comment]
Issue description is updated for marking them done.

**Comment by Prathamesh-007** [comment]
I am on MutableExceptionCheckExamplesTest

**Comment by Prathamesh-007** [comment]
Problem in MutableExceptionCheckExamplesTest example java classes.

Classes have same name which is causing ambiguity:
(Classes Example1.java and Example3.java)
<img width="960" alt="Duplicate Nested Classes" src="https://github.com/checkstyle/checkstyle/assets/77877305/dad4dc4b-d097-4043-9af6-0dbe29ff9428">

<br>
Renaming the classes to remove ambuguity. The remaining code is same:

(Classes Example1.java and Example2.java)
<img width="960" alt="Rename to avoid ambiguity" src="https://github.com/checkstyle/checkstyle/assets/77877305/e83af1cb-a935-4be3-a75f-a311179c99bc">

<br>
Is it ok if I rename the classes?

**Comment by MANISH-K-07** [comment]
@Prathamesh-007 , all such discussions and queries must be in PRs and not in issue page

**Comment by Acesif** [comment]
I am on IllegalIdentifierNameCheckExamplesTest

**Comment by Kexin2000** [comment]
I am on SeverityMatchFilterExamplesTest

**Comment by SteLeo1602** [comment]
I am on ReturnCountCheckExamplesTest

**Comment by SteLeo1602** [comment]
I am on IllegalThrowsCheckExamplesTest

**Comment by romani** [comment]
https://checkstyle.org/checks/sizes/linelength.html#Examples

There are only configs but no java code or any other languages code to show violation, what we should do is to add some code between such  comments and place in xdoc files macroses to render code part ( for now only configs are there)

**Comment by romani** [comment]
https://checkstyle.org/checks/sizes/linelength.html#Examples

There are only configs but no java code or any other languages code to show violation, what we should do is to add some code between such  comments and place in xdoc files macroses to render code part ( for now only configs are there).

If you think it is too complicated, please choose another Check for now.

**Comment by sushant1987** [comment]
@romani Thank you for the information.

**Comment by MohanadKh03** [comment]
I am on FinalLocalVariableCheck

**Comment by romani** [comment]
@MohanadKh03, please pull latest from our master, we merged big changes for this issue today, it will simplify your contribution

**Comment by chris-carneiro** [comment]
I'm on OneTopLevelClass 

Gentle notice, it looks like InterfaceIsTypeCheckExamplesTest.java was handled in this[PR](https://github.com/checkstyle/checkstyle/pull/14324)

**Comment by absolutely-sam** [comment]
I'm on RecordComponentNameCheck

**Comment by chris-carneiro** [comment]
I'm on ClassDataAbstractionCouplingCheck

**Comment by chris-carneiro** [comment]
TL;DR
The excludedPackages property behavior does not seem to match the documentation.

### Issue:
While testing the **ClassDataAbstractionCouplingCheck** against the **excludedPackages** property, I encountered a discrepancy with the behavior described in the documentation.

According to the documentation:
When a class is inside a package specified in the **excludedPackages** property, it should be excluded from the calculation of Data Abstraction Coupling (DAC).
This can be seen in the example of **Example11**, where the package **a.b** is expected to be excluded.

<img width="630" alt="image" src="https://github.com/user-attachments/assets/25193ab5-0784-4bb1-bf06-abf4afe7a630">

### Current Behavior:
However, classes from the excluded package are still being counted in the DAC, as illustrated by the following debug screenshots:

The classes from the excluded package are still considered "significant" and counted toward the DAC:
 `AbstractClassCouplingCheck#isSignificant` Line 434:8
<img width="1207" alt="image" src="https://github.com/user-attachments/assets/fe98b8fc-10c1-4188-a45d-2508699de168">

This behavior contradicts the expectation set by the documentation, where classes from the excluded package should be ignored.

### Next Steps:

- Should we update the documentation to reflect this behavior?
- Or, should this be raised as an issue to correct the check’s logic, so that classes in excluded packages are properly ignored?

Any insights?
@romani @stoyanK7

**Comment by romani** [comment]
@chris-carneiro, 

>does not seem to match the documentation.

I appreciate detailed investigation.
But let's just enable example and keep it to behave as it behaves now, even it is weird.
We will merge such update quickly.
After this we create issue on weird behavior of Check and we will review it, approve issue, handle it separately , in separate PR.

**Comment by AmitKumarDeoghoria** [comment]
GitHub, generate website

**Comment by SalmaAzeem** [comment]
I am working on TranslationCheckExamplesTest.java

**Comment by romani** [comment]
@SalmaAzeem, this Check is very complicated, feel free to choose different if you see it too complicated or create PR and we will guide you with help

**Comment by ironfist-15** [comment]
I am on translation check

**Comment by shaikhu** [comment]
I'd like to try and work on  ImportOrderCheckExamplesTest.

**Comment by Vignesh18280** [comment]
I'm working on BeforeExecutionExclusionFileFilterExamplesTest.

**Comment by ElinaZoldnere** [comment]
I'd like to try UniquePropertiesCheckExamplesTest.

**Comment by shaikhu** [comment]
@romani I think `RegexpSinglelineJavaCheckExamplesTest` can be marked as completed.

**Comment by ElinaZoldnere** [comment]
I'll take SuppressWithPlainTextCommentFilterExamplesTest

**Comment by Harsh-Agrawal96** [comment]
I am on RegexpCheckExamplesTest

**Comment by Rohan-here** [comment]
working on BeforeExecutionExclusionFileFilterExamplesTest.java

**Comment by Harsh-Agrawal96** [comment]
I am on SuppressionXpathSingleFilterExamplesTest

**Comment by shaikhu** [comment]
I'll take `SuppressWithNearbyCommentFilterExamplesTest`

**Comment by aboro-code** [comment]
Hi, if no one is working on SuppressionSingleFilterExamplesTest.java, I'd like to try it out.

**Comment by romani** [comment]
@aboro-code, 
Please work on it.

**Comment by ElinaZoldnere** [comment]
I'd like to try BeforeExecutionExclusionFileFilterExamplesTest

**Comment by romani** [comment]
@ElinaZoldnere , please do not hesitate to share PR, this filter might be tricky to enable. I will help as soon as PR is up.

**Comment by ahmedyoussefg** [comment]
I am on `SuppressionXpathSingleFilterExamplesTest.java`

**Comment by romani** [comment]
@zyadahmed , there is active PR for it https://github.com/checkstyle/checkstyle/pull/16173

**Comment by romani** [comment]
thanks a lot for participants on this issue !!!!!
it was more that 200 PRs, each PR has more than 20 files changed, it was huge work !!!!

this PR is great example of crowd sourcing.

**Comment by AmitKumarDeoghoria** [comment]
Glad to see it finally getting completed. Really grateful to have had the opportunity to work on this issue and contribute  50+ PRs on this. Learned a lot along the way, and it wouldn’t have been possible without the guidance and support from @romani and other mentors. Thanks for all the help!

**Comment by romani** [comment]
Special thanks to @AmitKumarDeoghoria , without such 50+ PR it will not be possible to resolve this issue this year.
